src/Stratis.Bitcoin.Features.PoA/PoAFeature.cs

```diff
@@ -0,0 +1,123 @@
+                features
+                    .AddFeature<PoAFeature>()
+                    .DependOn<ConsensusFeature>()
+                    .FeatureServices(services =>
+                    {
+                        //services.AddSingleton<  >(); //TODO PoA add miner and other things
+                    });
+            });
+
+            LoggingConfiguration.RegisterFeatureNamespace<ConsensusFeature>("consensus");
+            fullNodeBuilder.ConfigureFeature(features =>
+            {
+                features
+                    .AddFeature<ConsensusFeature>()
+                    .FeatureServices(services =>
+                    {
+                        services.AddSingleton<DBreezeCoinView>();
+                        services.AddSingleton<ICoinView, CachedCoinView>();
+                        services.AddSingleton<ConsensusController>();
+                        services.AddSingleton<IConsensusRuleEngine, PowConsensusRuleEngine>(); // TODO POA: PoA rule engine?
+                        services.AddSingleton<IChainState, ChainState>();
+                        services.AddSingleton<ConsensusQuery>()
+                            .AddSingleton<INetworkDifficulty, ConsensusQuery>(provider => provider.GetService<ConsensusQuery>())
+                            .AddSingleton<IGetUnspentTransaction, ConsensusQuery>(provider => provider.GetService<ConsensusQuery>());
+                        new PoAConsensusRulesRegistration().RegisterRules(fullNodeBuilder.Network.Consensus);
+                    });
+            });
+
+            return fullNodeBuilder;
+        }
+
+        private class PoAConsensusRulesRegistration : IRuleRegistration
+        {
+            public void RegisterRules(IConsensus consensus)
+            {
+                consensus.HeaderValidationRules = new List<IHeaderValidationConsensusRule>()
+                {
+                    new HeaderTimeChecksPoARule(),
+                    new StratisHeaderVersionRule(),
+                    new PoAHeaderSignatureRule()
+                };
+
+                consensus.IntegrityValidationRules = new List<IIntegrityValidationConsensusRule>()
+                {
+                    new BlockMerkleRootRule(),
+                    new PoAIntegritySignatureRule()
+                };
+
+                consensus.PartialValidationRules = new List<IPartialValidationConsensusRule>()
+                {
+                    new SetActivationDeploymentsPartialValidationRule(),
+
+                    // rules that are inside the method ContextualCheckBlock
+                    new TransactionLocktimeActivationRule(), // implements BIP113
+                    new CoinbaseHeightActivationRule(), // implements BIP34
+                    new BlockSizeRule(),
+
+                    // rules that are inside the method CheckBlock
+                    new EnsureCoinbaseRule(),
+                    new CheckPowTransactionRule(),
+                    new CheckSigOpsRule(),
+                };
```

Comment:
doesn't match with research document
PV rules:

> SetActivationDeploymentsPartialValidationRule
> TransactionLocktimeActivationRule
> BlockSizeRule
> CheckPowTransactionRule
> CheckSigOpsRule

Reply:
Yes, I have 2 more: 
CoinbaseHeightActivationRule
EnsureCoinbaseRule

forgot to update PoA document after deciding that coinbase tx should be in the block even for PoA. So it's a problem in the document- updated it already.